java/client/src/org/openqa/selenium/safari/SafariDriver.java

```diff
@@ -139,4 +147,138 @@ protected void stopClient() {
     // ... and convert it.
     return target.convertFromBase64Png(base64);
   }
+
+  @Override
+  public TargetLocator switchTo() {
+    if (this.getCommandExecutor() instanceof SafariDriverCommandExecutor) {
+      return new LegacyRemoteTargetLocator();
+    }
+
+    return super.switchTo();
+  }
+
+  @Override
+  public Options manage() {
+    if (this.getCommandExecutor() instanceof SafariDriverCommandExecutor) {
+      return new LegacyRemoteWebDriverOptions();
+    }
+
+    return super.manage();
+  }
+
+
+  protected class LegacyRemoteTargetLocator extends RemoteTargetLocator {
```

Comment:
Or we make it public, like the superclasses? I personally avid restricting the access if it's not required due to internal private dependencies, since it takes all ability to "fix" behavior in subclasses (as we're doing here).